### PR TITLE
Mauro oto/comprehensive guide to imap interaction with ruby

### PR DESCRIPTION
**Pull request by mauro-oto**
Hey @etagwerker, 

This is a blog post which sort of condenses our experience with IMAP. 

Please check it out, thanks!

### CODE REVIEW

File: _posts/2016-05-13-a-comprehensive-guide-to-interacting-with-imap-using-ruby.markdown [hunk 1]
```diff
@@ -0,0 +1,260 @@
+---
+layout: post
+title: "A comprehensive guide to interacting with IMAP using Ruby"
+date: 2016-05-20 16:56:00
+categories: ["ruby", "imap"]
+author: "mauro-oto"
+---
+
+A few times in the past I've had to interact with IMAP via Ruby, and wrapping
+your head around its API is not so easy. Not only is the IMAP API a bit obscure
```
Comment (etagwerker) [inline _posts/2016-05-13-a-comprehensive-guide-to-interacting-with-imap-using-ruby.markdown]: @mauro-oto You could probably link to it: https://tools.ietf.org/html/rfc3501

File: _posts/2016-05-13-a-comprehensive-guide-to-interacting-with-imap-using-ruby.markdown [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+=> #<Net::IMAP:0x007fa1c77064b0 ...
+```
+
+which skips SSL verification. Note that while this is probably not such a good
+idea, even the most popular Gmail Ruby gem,
+[Gmail](https://github.com/gmailgem/gmail/blob/353ddcc8cc0c5b57ad1d3a412f11365ccc12b7d6/lib/gmail/client/base.rb#L26),
+does this.
+
+After the connection is established, you can login using a couple of different
+ways. First, the most common way, using the email and the password:
```
Comment (etagwerker) [inline _posts/2016-05-13-a-comprehensive-guide-to-interacting-with-imap-using-ruby.markdown]: @mauro-oto it should say "using email and password"

File: _posts/2016-05-13-a-comprehensive-guide-to-interacting-with-imap-using-ruby.markdown [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+```ruby
+[7] pry(main)> imap.select("[Gmail]/All Mail")
+=> #<struct Net::IMAP::TaggedResponse
+ name="OK",
+ raw_data="RUBY0015 OK [READ-WRITE] [Gmail]/All Mail selected. (Success)\r\n">
+```
+
+## Searching email messages
+
+After selecting an email box, you can search email messages and return either
```
Comment (etagwerker) [inline _posts/2016-05-13-a-comprehensive-guide-to-interacting-with-imap-using-ruby.markdown]: @mauro-oto it should say "you can search email messages and find either ... "

File: _posts/2016-05-13-a-comprehensive-guide-to-interacting-with-imap-using-ruby.markdown [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+the sequence number (`seqno`) for the email, or the unique id (`uid`).
```
Comment (etagwerker) [inline _posts/2016-05-13-a-comprehensive-guide-to-interacting-with-imap-using-ruby.markdown]: @mauro-oto it should say "the sequence number (`seqno`) or unique id (`uid`) for the email."

File: _posts/2016-05-13-a-comprehensive-guide-to-interacting-with-imap-using-ruby.markdown [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+# From within Gmail, we delete one of our very first emails in the All Mail box and then make the same search we did before.
+
+[11] pry(main)> imap.uid_search(["SUBJECT", "Jedi"])
+=> [23]
+```
+
+## Fetching email messages
+
+After we selected a mailbox and we know a message's `seqno` or `uid`, we can
+fetch the contents of the email. If you know the `seqno`:
```
Comment (etagwerker) [inline _posts/2016-05-13-a-comprehensive-guide-to-interacting-with-imap-using-ruby.markdown]: @mauro-oto You talk about the contents but later you fetch just the `ENVELOPE` - You might want to rephrase the sentence above ^^

**Comment by etagwerker** [comment]
@mauro-oto Great! Please update the date in the post and update the PR so that I can merge it.

**Comment by mauro-oto** [comment]
@etagwerker done!

**Pull request merged**